pkg/sql/catalog/internal/catkv/catalog_reader.go

```diff
@@ -225,6 +228,11 @@ func (cr catalogReader) GetDescriptorEntries(
 			for _, id := range ids {
 				if id != descpb.InvalidID && mc.LookupDescriptorEntry(id) == nil {
 					b.Get(catalogkeys.MakeDescMetadataKey(codec, id))
+					// Fetch all the comments on the descriptor in same batch.
+					for _, t := range keys.AllCommentTypes {
+						cmtKeyPrefix := catalogkeys.MakeObjectCommentsMetadataPrefix(codec, t, id)
+						b.Scan(cmtKeyPrefix, cmtKeyPrefix.PrefixEnd())
+					}
```

Comment:
This is probably out of scope, still, should we be doing the same thing for namespace entries?

Reply:
yes/no, we may just lazily use descriptors, either read or write of namespace can really be delegate to descriptors (uncommitted descriptors), that way we may just simply add methods to `Collection`. The downsize is that the pattern changes from comments and zoneconfigs. Maybe there is some blindspot until I make progress to that point. But this sounds like feasible.